hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1694,16 +1694,20 @@ private void deleteBlob(final Path sourcePath,
   }
 
   public void delete(final Path path, final boolean recursive,
-      TracingContext tracingContext) throws AzureBlobFileSystemException {
+      TracingContext tracingContext) throws IOException {
+    LOG.debug("delete filesystem: {} path: {} recursive: {}",
+        client.getFileSystem(),
+        path,
+        String.valueOf(recursive));
+    if (getPrefixMode() == PrefixMode.BLOB) {
```

Comment:
Should have a fallback mechanism like other API's

Reply:
kindly clarify on what fallback mechanism you are referring to. Thanks.